src/api/course/index.ts

```diff
@@ -320,13 +321,38 @@ export class CourseApi {
     return _.get((this.app.signalk as any).self, 'navigation.position')
   }
 
-  private validateCourseInfo(info: CourseInfo) {
+  private async validateCourseInfo(info: CourseInfo) {
     if (
       typeof info.activeRoute !== 'undefined' &&
       typeof info.nextPoint !== 'undefined' &&
       typeof info.previousPoint !== 'undefined'
     ) {
-      return info
+      // validate waypoint href
+      if (!info.activeRoute && info.nextPoint?.href) {
```

Comment:
This section of conditional logic is really hard to follow, at least for me. 

Could we instead have a series of self documenting test functions

```
if (isValidWaypointCourse(info)) {
  return info
}
....
debug(`** Error: Loaded course data is invalid!! (using default) **`)
return this.courseInfo
```

Reply:
Have created test functions.
Also included pointindex check for active route.